### PR TITLE
Report not ready when no namespace matches

### PR DESCRIPTION
**Pull request by shuheiktgw**
## Problem Statement

The details are described in the issue below, and I believe ClusterExternalSecret should be not ready if no namespace matches. Thank you for your review 🙂  

## Related Issue

Closes https://github.com/external-secrets/external-secrets/issues/2507

## Proposed Changes

SSIA

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go [hunk 1]
```diff
@@ -92,14 +93,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	labelSelector, err := metav1.LabelSelectorAsSelector(&clusterExternalSecret.Spec.NamespaceSelector)
 	if err != nil {
 		log.Error(err, errConvertLabelSelector)
-		return ctrl.Result{RequeueAfter: refreshInt}, err
```
Comment (shuheiktgw) [inline pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go:95]: `RequeueAfter` is ignored if we return an error, so I just removed it.

File: pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller_test.go [hunk 2]
```diff
@@ -361,31 +361,38 @@ var _ = Describe("ClusterExternalSecret controller", func() {
 				}
 			},
 		}),
-		Entry("Should crate an external secret and if one with the same name has been deleted", testCase{
+		Entry("Should crate an external secret if one with the same name has been deleted", testCase{
```
Comment (shuheiktgw) [inline pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller_test.go:364]: Fix this flaky test, which seems to have the same problem with https://github.com/external-secrets/external-secrets/pull/2558 🙇 

https://github.com/external-secrets/external-secrets/actions/runs/5774855175/job/15652226003?pr=2582

File: pkg/controllers/clusterexternalsecret/util.go [hunk 3]
```diff
@@ -22,17 +22,31 @@ import (
...
+		return &esv1beta1.ClusterExternalSecretStatusCondition{
+			Type:   esv1beta1.ClusterExternalSecretReady,
+			Status: v1.ConditionTrue,
+		}
+	case len(failedNamespaces) < len(namespaceList.Items):
+		return &esv1beta1.ClusterExternalSecretStatusCondition{
+			Type:    esv1beta1.ClusterExternalSecretPartiallyReady,
+			Status:  v1.ConditionTrue,
+			Message: errNamespacesFailed,
+		}
```
Comment (moolen) [inline pkg/controllers/clusterexternalsecret/util.go:42]:
```suggestion
	switch {
	case len(namespaceList.Items) == 0:
		return &esv1beta1.ClusterExternalSecretStatusCondition{
			Type:    esv1beta1.ClusterExternalSecretReady,
			Status:  v1.ConditionFalse,
			Message: errNamespaceNotFound,
		}
	case len(failedNamespaces) == 0:
		return &esv1beta1.ClusterExternalSecretStatusCondition{
			Type:   esv1beta1.ClusterExternalSecretReady,
			Status: v1.ConditionTrue,
		}
	case len(failedNamespaces) < len(namespaceList.Items):
		return &esv1beta1.ClusterExternalSecretStatusCondition{
			Type:    esv1beta1.ClusterExternalSecretPartiallyReady,
			Status:  v1.ConditionTrue,
			Message: errNamespacesFailed,
		}
```

I find it superfluous that we have both a `Ready` and `NotReady` condition. 

With a bit fumbling around i ended up in a status of:
1. deploy ces with a matching selector
2. change selector to something that doesn't match
```yaml
status:
  conditions:
  - status: "True"
    type: Ready
  - message: no namespace matches
    status: "True"
    type: NotReady
```

Which doesn't make sense. I'd remove the `NotReady` condition type and instead work with the `Ready` condition and set it to false. That way we get a consistent status. WDYT?

Comment (shuheiktgw) [reply pkg/controllers/clusterexternalsecret/util.go:42]: Good catch, thank you! 🙇 
I agree that the current design is a little too complicated, which makes it error-prone. Let me fix that!

Comment (shuheiktgw) [reply pkg/controllers/clusterexternalsecret/util.go]: Should we remove `PartiallyReady` as well? It seems if we substitute `NotReady & true` with `Ready & false`, `PartiallyReady & true` is not mutually exclusive with `Ready& false` anymore. So I personally think it makes more sense if we just remove `PartiallyReady` and use `Ready & false` for that case as well 🤔 

I've fixed the PR that way, please let me know what you think about that 🙇

Comment (moolen) [reply pkg/controllers/clusterexternalsecret/util.go]: Agree, that makes sense! I'll do another round of manual testing in the next days.

**Review by moolen — APPROVED** [review]
nod
LGTM, that's a great improvement, thank you!

**Pull request merged**